test/cli_util_test.dart

```diff
@@ -38,8 +39,15 @@ void defineTests() {
   });
 
   group('applicationConfigHome', () {
-    test('returns a string', () {
-      expect(applicationConfigHome('dart'), isA<String>());
+    test('returns a non-empty string', () {
+      expect(applicationConfigHome('dart'), isNotEmpty);
+    });
+
+    test('has an ancestor folder that exists', () {
+      final path = p.split(applicationConfigHome('dart'));
+      // We expect that first two segments of the path exists.. This is really
+      // just a dummy check that some part of the path exists.
+      expect(Directory(p.joinAll(path.take(2))).existsSync(), isTrue);
```

Comment:
This is much better.

For thorough test we could consider running something along:
```
File('script.dart').writeAsStringAsync('''
import 'package:cli_util';
main() => print(applicationConfigHome('dart');
''');
final result = Process.runSync(Platform.resolvedExecutable, ['script.dart'], environment: {
  if (Platform.isLinux) 'CDG_CONFIG_HOME': '....'
  ....
});
  expect(result.stdout, ...);
```

Not sure it is worth it.

Reply:
Yeah, that is an option.